core/src/main/java/org/neo4j/ogm/metadata/ObjectAnnotations.java

```diff
@@ -31,11 +36,38 @@
 /**
  * @author Vince Bickers
  * @author Gerrit Meier
+ * @author Michael J. Simons
  */
 public class ObjectAnnotations {
 
-    private final Map<String, AnnotationInfo> annotations = new HashMap<>();
+    static ObjectAnnotations of(Annotation... annotations) {
+
+        Map<String, AnnotationInfo> annotationInfo = Arrays.stream(annotations) //
+            .map(AnnotationInfo::new) //
+            .collect(toMap(AnnotationInfo::getName, Function.identity()));
+        return new ObjectAnnotations(annotationInfo);
+    }
+
+    private final Map<String, AnnotationInfo> annotations;
```

Comment:
Even if the above is a static method I would, at least, suggest to move this above. The other class that gets the `of` builder method also keeps it fields on top.